src/Lean/PrettyPrinter/Delaborator/Options.lean

```diff
@@ -125,6 +125,16 @@ register_builtin_option pp.instanceTypes : Bool := {
   group    := "pp"
   descr    := "(pretty printer) when printing explicit applications, show the types of inst-implicit arguments"
 }
+register_builtin_option pp.omitDeepTerms : Bool := {
+  defValue := false
+  group    := "pp"
+  descr    := "(pretty printer) omit deeply nested terms, replacing them with `⋯`"
+}
+register_builtin_option pp.maxTermDepth : Nat := {
```

Comment:
Marc and I were discussing renaming this for discoverability. Some options:
* `pp.omitDeepTerms.threshold`, which I think is a nice name for the concept.
* `pp.omitDeepTerms.maxDepth`, which is parallel to `pp.raw.maxDepth`, but it's not strictly a *max* depth in this case due to the shallow terms heuristic.

Reply:
Either is probably fine, if you need comments from the peanut gallery: I buy your arguments in favor of `threshold`